packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -450,6 +452,11 @@ export default class CodeGen {
             }
         }
 
+        // TODO [#3370]: remove experimental template expression flag
+        if (this.state.config.experimentalComplexExpressions) {
+            return bindComplexExpression(expression as ComplexExpression, this);
```

Comment:
The expression parsed AST is reused if an expression is duplicated in the template. If an expression appears twice in the template, the second occurrence will be bound twice to the component which is incorrect.

```html
<template>
    <x-pert>{`${name}: ${occupation}`}</x-pert>
    <x-pert>{`${name}: ${occupation}`}</x-pert>
</template>
```

Reply:
Ack. This was a mistake, will be fixed. Thanks for the catch!